### PR TITLE
add BIOS and IPMI firmware version to node properties

### PR DESCRIPTION
**Pull request by berlincount**
Collects IPMI/iDRAC and BIOS versions for Genesis as well.

MegaRaid FW / Device info also needs to be collected, PR for that should follow tomorrow.

Relates to https://github.com/Shopify/datastores/issues/1987

### CODE REVIEW

File: spec/collector_spec.rb [hunk 1]
```diff
@@ -123,6 +125,8 @@
         stub_dmi('baseboard-product-name', 'ABC456B+')
         stub_dmi('baseboard-serial-number', '34524623454')
         stub_dmi('chassis-serial-number', '2376482364')
+        stub_dmi('bios-version', '4.23')
+        stub_shellout('ipmitool mc info', fixture('ipmitool_mc_info'))
```
Comment (dalehamel) [inline spec/collector_spec.rb]: just a heads-up that ipmitool can be flakey, might want additional error handling

**Comment by dalehamel** [comment]
Looks pretty good to me, really glad you added tests too :heart: 
@dwradcliffe as I understand it, genesis should just ignore any data it doesn't know how to parse, right?

**Comment by dwradcliffe** [comment]
> genesis should just ignore any data it doesn't know how to parse, right?

correct

**Comment by dwradcliffe** [comment]
Oh but it does know how to handle `properties`. So these will all be saved.

**Comment by dalehamel** [comment]
oh sweet, so then it'll actually display them already!

File: spec/collector_spec.rb (hunk 1 above)
Comment (berlincount) [inline spec/collector_spec.rb]: yeah, it seems to depend on which parts you query - "mc info" seems to be safe, and the code collecting in genesis_collector seems to be safeguarding already.

**Comment by berlincount** [comment]
I consider reasonable tests non-optional ;) If you're happy with it, please mark reviewed and merge already :)

Stuff shall show up under properties, then.

**Comment by berlincount** [comment]
MegaCli is a different beast, it will need separate viewing and bit more intricate parsing - nothing that can't be solved in an hour, but .. they've basically got interesting stuff like "a running FirmWare version" and "an uploaded and ready to go firmware version", which show up the same - just in separate sections of the non-computerparsable output. boo.

while I'm at it, I'd also add dumping the info of the Logical Drives (RAID level, status), which seem to have caused irritation in some places already (backupservers ...), so I figure a separate PR is warranted.

**Comment by jordanwheeler** [comment]
yeah let's keep those PRs separate.

good stuff @berlincount

**Comment by berlincount** [comment]
Brrr. https://shipit.shopify.io/shopify/genesis_collector/rubygems has a lot of undeployed commits.

@dturn @dwradcliffe @dalehamel are you comfortable with me pulling the trigger here?

**Comment by dwradcliffe** [comment]
The shipit deploy doesn't work, we've just been publishing the gem manually. Feel free to try to fix it or we can just publish.

**Comment by berlincount** [comment]
Ah. Oh. Hm. Please publish :)

**Comment by berlincount** [comment]
deploys work again, the git cache was fscked up

**Pull request merged**